test/domains.jl

```diff
@@ -142,61 +109,67 @@ end
-        @test coordinates(domain).subsurface === coordinates(column)
-        @test obtain_surface_space(column.space) === point.space
+        @test typeof(column) == Column{FT}
+        @test typeof(point) == Point{FT}
+        @test parent(coordinates(point)) == parent(coordinates(domain).surface)
+        @test parent(coordinates(domain).subsurface) ==
+              parent(coordinates(column))
     end
 end
 
-
-@testset "LSMMultiColumnDomain" begin
-    for FT in TestFloatTypes
-        zmin = FT(1.0)
-        zmax = FT(2.0)
-        zlim = FT.((zmin, zmax))
-        xlim = FT.((0.0, 10.0))
-        ylim = FT.((0.0, 1.0))
-        zlim = FT.((zmin, zmax))
-        nelements = (1, 1, 5)
-        npolynomial = 2
-        domain = LSMMultiColumnDomain(;
-            xlim = xlim,
-            ylim = ylim,
-            zlim = zlim,
-            nelements = nelements,
-            periodic = (true, true),
-            npolynomial = npolynomial,
-        )
-        plane = domain.surface
-        box = domain.subsurface
-
-        @test typeof(box) == HybridBox{FT, typeof(box.space)}
-        @test typeof(plane) == Plane{FT, typeof(plane.space)}
-        @test coordinates(plane) === coordinates(domain).surface
-        @test coordinates(domain).subsurface === coordinates(box)
-        @test obtain_surface_space(box.space) === plane.space
-    end
-end
-
-
-
-@testset "LSMSphericalShellDomain" begin
+TestFloatTypes = (Float32, Float64)
+@testset "PlantHydraulicsDomain" begin
     for FT in TestFloatTypes
-        domain = LSMSphericalShellDomain(;
-            radius = FT(100.0),
-            height = FT(30.0),
-            nelements = (6, 20),
-            npolynomial = 3,
+        Δz = 1.0
+        roots = [FT(1.0), FT(2.0), FT(3.0)]
+        stem = Int64(5)
+        leaves = Int64(4)
+        comp_points = Vector(FT(0.5):FT(1.0):FT(8.5))
+        top_of_compartments = Vector(FT(0.0):FT(1.0):FT(9.0))
+        comp_labels =
+            [:stem, :stem, :stem, :stem, :stem, :leaf, :leaf, :leaf, :leaf]
+        test_tuple = (root = 1, stem = 2, leaf = 3)
+        @test test_tuple[:root] == 1
+
+        #test addition of surfaces
+
+        testing = PlantHydraulicsDomain(roots, stem, leaves, FT(Δz))
+        @test testing.root_depths == [1.0, 2.0, 3.0]
+        @test testing.n_stem == 5
+        @test testing.n_leaf == 4
+        @test testing.compartment_midpoints == comp_points
+        @test testing.compartment_surfaces == top_of_compartments
+        @test testing.compartment_labels == comp_labels
+        for i in 1:5
+            @test test_tuple[testing.compartment_labels[i]] == 2
+        end
+        for i in 6:9
+            @test test_tuple[testing.compartment_labels[i]] == 3
+        end
+
+        testing2 = PlantHydraulicsDomain(
+            roots,
+            stem,
+            leaves,
+            comp_points,
+            top_of_compartments,
         )
-        surf = domain.surface
```

Comment:
lots of code deleted in this file that should be here. the only change your code should bring compared with the version in main is the change in the plant hydraulics domain test code.